File: backend/app/schemas/semester.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class SemesterBase(BaseModel):
    academic_year_id: int
    number: int
    is_active: bool = True


class SemesterCreate(BaseModel):
    academic_year_id: int
    number: int


class SemesterRead(SemesterBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class SemesterUpdate(BaseModel):
    academic_year_id: int | None = None
    number: int | None = None
    is_active: bool | None = None

File: frontend/src/types/semester.ts
// Mirrors backend/app/schemas/semester.py in full.

export interface Semester {
  id: number;
  academic_year_id: number;
  number: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface SemesterCreate {
  academic_year_id: number;
  number: number;
}

export interface SemesterUpdate {
  academic_year_id?: number;
  number?: number;
  is_active?: boolean;
}
